modyn/storage/src/internal/file_wrapper/binary_file_wrapper.cpp

```diff
@@ -61,13 +58,25 @@ std::vector<int64_t> BinaryFileWrapper::get_all_labels() {
   std::vector<int64_t> labels = std::vector<int64_t>();
   labels.reserve(num_samples);
 
-  for (uint64_t i = 0; i < num_samples; ++i) {
-    get_stream()->seekg(static_cast<int64_t>(i * record_size_), std::ios::beg);
-
-    std::vector<unsigned char> label_vec(label_size_);
-    get_stream()->read(reinterpret_cast<char*>(label_vec.data()), static_cast<int64_t>(label_size_));
+  std::vector<unsigned char> label_vec(label_size_);
 
-    labels.push_back(int_from_bytes(label_vec.data(), label_vec.data() + label_size_));
+  // Duplicating the loop here to avoid doing the endianess check in every loop iteration;
+  // there might be more idomatic ways to do this though.
+  if (this->little_endian_) {
+    for (uint64_t i = 0; i < num_samples; ++i) {
+      get_stream()->seekg(static_cast<int64_t>(i * record_size_), std::ios::beg);
+      get_stream()->read(reinterpret_cast<char*>(label_vec.data()), static_cast<int64_t>(label_size_));
+      labels.push_back(int_from_bytes_little_endian(label_vec.data(), label_vec.data() + label_size_));
+      label_vec.clear();
+    }
+  }
+  else {
+    for (uint64_t i = 0; i < num_samples; ++i) {
+      get_stream()->seekg(static_cast<int64_t>(i * record_size_), std::ios::beg);
+      get_stream()->read(reinterpret_cast<char*>(label_vec.data()), static_cast<int64_t>(label_size_));
+      labels.push_back(int_from_bytes_big_endian(label_vec.data(), label_vec.data() + label_size_));
+      label_vec.clear();
+    }
```

Comment:
so I guess previously in `get_all_labels` they even didn't take endianness into account right? Thanks for the catch!

Reply:
Previously get_all_labels was agnostic of the `endianness` and relied on `int_from_bytes` to handle it. I could have done it there and let get_all_labels the same. However, I guess it makes more sense to not do the check for `byteorder` in every iteration of the for loop (potentially every 4-byte block) read-in.